src/cyclotom.c

```diff
@@ -1172,24 +1172,17 @@ static Obj ProdCycInt(Obj opL, Obj opR)
 
     /* otherwise multiply every coefficent                                 */
     else {
-        hdP = NewBag( T_CYC, SIZE_CYC(opL) * (sizeof(Obj)+sizeof(UInt4)) );
```

Comment:
Can you give me a feeling what's changed here, the code seems to have entirely changed.

Reply:
So the static analyzer complained about the values `exs` and `exp` in the old lines 1179 and 1181. I at first just removed them, but then had a closer look at the code. It turns out that it copies half the data; and the other half are coefficients which are multiplied by a fixed scalar. So I changed the code to first copy everything via a single `memcpy`, and then update the scalars in-place. This way, we only need to compute one bag pointer per iteration (instead of 4), and it also is conceptually easier to understand, IMHO. I think this all gets clearer if one doesn't use the "unified diff" view on GitHub, but rather compares old and new code side-by-side.